addons/web/controllers/utils.py

```diff
@@ -199,6 +200,10 @@ def _get_login_redirect_url(uid, redirect=None):
     return parsed.replace(query=werkzeug.urls.url_encode(qs)).to_url()
 
 
+def is_user_internal(uid):
+    return request.env['res.users'].sudo().browse(uid)._is_internal()
```

Comment:
We can't read `res.users`'s `share` field as public user, and we chose to use `share` instead of `has_group('base.group_internal')` to use the cache. (also not sudo inside `_is_internal()` as it is currently only required  for the login use case, but I guess it's open to discussion/change).

Reply:
I don't understand your comment.

`_is_internal()` currently does use sudo internally, so it always works, without the need for an outer sudo. Pick one or the other, both is unnecessary.

Reply:
My mistake, we had indeed changed that. Thanks!